packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2305,13 +2306,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       // export default interface allowed in:
       // https://github.com/Microsoft/TypeScript/pull/16040
       if (this.state.value === "interface") {
```

Comment:
We should avoid using `lookahead()` as it is expensive. We may have to save the start location of the `interface` token, call `this.next()` and then check if the current token is a `tt.name`.

Reply:
> We can perform lookahead after interface and see if it is a tt.name in like how we handle export default abstract class: https://github.com/babel/babel/issues/13604

Reply:
Probably it was suggested because `lookahead` has recently got perf improvements. I would still avoid it if we can.

Reply:
I mentioned `lookahead` before because it could solve the original problem. In TS the `interface` is a reserved keyword so we can avoid the lookahead and just throw if there is no `tt.name` after `interface`, since

```js
export default interface[0];
export default interface(0);
```

are invalid in TS, though they are valid in JS.

Reply:
Well, `interface` is a reserved word in JS strict mode and thus in modules (https://tc39.es/ecma262/#sec-identifiers-static-semantics-early-errors, last production).

Reply:
Since export default is only allowed in module, we can safely assume interface is a reserved word here.